q2_feature_table/_filter.py

```diff
@@ -6,16 +6,118 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
+import sqlite3
+
 import biom
-import skbio
+import qiime
+import numpy as np
+
+
+def _ids_where(metadata, where):
+    df = metadata.to_dataframe()
+
+    conn = sqlite3.connect(':memory:')
+    conn.row_factory = lambda cursor, row: row[0]
+
+    df.to_sql('metadata', conn)
+
+    c = conn.cursor()
+    c.execute('SELECT * FROM metadata ORDER BY ROWID ASC LIMIT 1;')
+    names = [description[0] for description in c.description]
+    id_column = names[0]
```

Comment:
Unfortunately there is no guarantee the the first column is the index column, or even that the column order will always be the same across queries. Probably makes sense to drop L24-L27 and replace with something like:

```python
id_column = df.index.name
```